File: examples/python/remove_layer.py
from gi.repository import GES, Gst

from create_timeline_with_n_layers import create_timeline_with_n_layers
from add_one_clip_in_each_layer import add_one_clip_in_each_layer
from add_one_video_track import add_one_video_track
from play_timeline import play_timeline

if __name__=="__main__":
    Gst.init([])
    GES.init()

    timeline = create_timeline_with_n_layers(2)
    add_one_clip_in_each_layer(timeline, 10 * Gst.SECOND)
    add_one_video_track (timeline)
    for layer in timeline.get_layers():
        timeline.remove_layer (layer)

    timeline.commit()
    # This will return immediately, as the timeline is now empty
    play_timeline (timeline)

File: examples/c/remove_layer.c
#include "examples.h"

int
main (int argc, char **argv)
{
  GList *tmp, *layers;
  GESTimeline *timeline;

  gst_init (NULL, NULL);
  ges_init ();

  timeline = create_timeline_with_n_layers (2);
  add_one_clip_in_each_layer (timeline, 10 * GST_SECOND);
  add_one_video_track (timeline);

  layers = ges_timeline_get_layers (timeline);

  for (tmp = layers; tmp; tmp = tmp->next)
    ges_timeline_remove_layer (timeline, GES_LAYER (tmp->data));

  g_list_free (layers);

  ges_timeline_commit (timeline);

  /* This will return immediately, as the timeline is now empty */
  play_timeline (timeline);

  return 0;
}
